modules/47_regipol/regiCarbonPrice/bounds.gms

```diff
@@ -18,36 +18,116 @@
+                                            /  pm_prodCouple(regi,"pegas","seel","gaschp","sehe")
+*** divide by capacity factor to get from seel production in TWa to capacity in TW
+                                            / pm_cf(t,regi,"gaschp");
+
+        vm_cap.up(t,regi,"gaschp","1")= 0.55
+*** total FE heat demand from industry and buildings calibration trajectories
+                                            * (pm_cesdata(t,regi,"feheb","quantity")
+                                                + pm_cesdata(t,regi,"feheb","quantity"))
+*** divide by tdhes efficiency get from FE heat to SE heat, i.e. take into account heat transmission losses
+                                            / pm_eta_conv(t,regi,"tdhes")
+*** divide by pm_prodCouple to get from heat production to production of seel
+                                            /  pm_prodCouple(regi,"pegas","seel","gaschp","sehe")
+*** divide by capacity factor to get from seel production in TWa to capacity in TW
+                                            / pm_cf(t,regi,"gaschp");
+
+        vm_cap.lo(t,regi,"biochp","1")= 0.15
+*** total FE heat demand from industry and buildings calibration trajectories
+                                            * (pm_cesdata(t,regi,"feheb","quantity")
+                                                + pm_cesdata(t,regi,"feheb","quantity"))
+*** divide by tdhes efficiency get from FE heat to SE heat, i.e. take into account heat transmission losses
+                                            / pm_eta_conv(t,regi,"tdhes")
+*** divide by pm_prodCouple to get from heat production to production of seel
+                                            /  pm_prodCouple(regi,"pebiolc","seel","biochp","sehe")
+*** divide by capacity factor to get from seel production in TWa to capacity in TW
+                                            / pm_cf(t,regi,"biochp");
+
+        vm_cap.up(t,regi,"biochp","1")= 0.25
+*** total FE heat demand from industry and buildings calibration trajectories
+                                            * (pm_cesdata(t,regi,"feheb","quantity")
+                                                + pm_cesdata(t,regi,"feheb","quantity"))
+*** divide by tdhes efficiency get from FE heat to SE heat, i.e. take into account heat transmission losses
+                                            / pm_eta_conv(t,regi,"tdhes")
+*** divide by pm_prodCouple to get from heat production to production of seel
+                                            /  pm_prodCouple(regi,"pebiolc","seel","biochp","sehe")
+*** divide by capacity factor to get from seel production in TWa to capacity in TW
+                                            / pm_cf(t,regi,"biochp");
+
+
+    );
 );
 
+*' ####### Carbon Management
+
+*' This limits CO2 underground injection up to 2030 in line with recent developments as of 2023. 
+vm_co2CCS.up(t,regi,"cco2","ico2",te,rlf)$((t.val le 2030) AND (sameas(regi,"DEU"))) = 1e-3;
```

Comment:
Anne added limits to vm_co2CCS in the core code.
This should be probably handled in the `p_boundCapCCS.cs4r` file directly to avoid two parts of the code overwriting one another.

I am fine with merging this way, as long as you add a comment to this stating that, and make sure Anne is aware of including this German specific data to her formulation.

Reply:
Thanks. Anne actually removed it already so I also remove it here.